core/src/integration-test/java/com/scalar/db/api/DistributedTransactionAdminRepairTableIntegrationTestBase.java

```diff
@@ -0,0 +1,175 @@
+
+  protected void initialize() throws Exception {}
+
+  protected abstract Properties getProperties();
+
+  protected Properties getStorageProperties() {
+    return getProperties();
+  }
+
+  protected String getNamespace() {
+    return NAMESPACE;
+  }
+
+  protected String getTable() {
+    return TABLE;
+  }
+
+  private void createTable() throws ExecutionException {
+    Map<String, String> options = getCreateOptions();
+    admin.createNamespace(getNamespace(), options);
+    admin.createTable(getNamespace(), getTable(), TABLE_METADATA, options);
+    admin.createCoordinatorTables(true, options);
+  }
+
+  protected Map<String, String> getCreateOptions() {
+    return Collections.emptyMap();
+  }
+
+  private void dropTable() throws ExecutionException {
+    admin.dropTable(getNamespace(), TABLE);
+    admin.dropNamespace(getNamespace());
+    admin.dropCoordinatorTables(true);
+  }
+
+  @BeforeAll
+  protected void beforeAll() throws Exception {}
+
+  @BeforeEach
+  protected void beforeEach() throws Exception {
+    TransactionFactory factory =
+        TransactionFactory.create(TestUtils.addSuffix(getProperties(), TEST_NAME));
+    admin = factory.getTransactionAdmin();
+    createTable();
+    adminTestUtils = AdminTestUtils.create(TestUtils.addSuffix(getStorageProperties(), TEST_NAME));
+  }
+
+  @AfterEach
+  protected void afterEach() throws Exception {
+    dropTable();
+  }
+
+  @AfterAll
+  protected void afterAll() throws Exception {}
+
+  @Test
+  public void repairTableAndCoordinatorTable_ForDeletedMetadataTable_ShouldRepairProperly()
+      throws Exception {
+    // Arrange
+    adminTestUtils.dropMetadataTable();
+
+    // Act
+    admin.repairTable(getNamespace(), getTable(), TABLE_METADATA, getCreateOptions());
+    admin.repairCoordinatorTables(getCreateOptions());
+
+    // Assert
+    assertThat(admin.tableExists(getNamespace(), TABLE)).isTrue();
+    assertThat(admin.getTableMetadata(getNamespace(), TABLE)).isEqualTo(TABLE_METADATA);
+    assertThat(admin.coordinatorTablesExist()).isTrue();
+    assertTableMetadataForCoordinatorTableArePresent();
+  }
+
+  @Test
+  public void repairTableAndCoordinatorTable_ForTruncatedMetadataTable_ShouldRepairProperly()
+      throws Exception {
+    // Arrange
+    adminTestUtils.truncateMetadataTable();
+
+    // Act
+    admin.repairTable(getNamespace(), getTable(), TABLE_METADATA, getCreateOptions());
+    admin.repairCoordinatorTables(getCreateOptions());
+
+    // Assert
+    assertThat(admin.tableExists(getNamespace(), TABLE)).isTrue();
+    assertThat(admin.getTableMetadata(getNamespace(), TABLE)).isEqualTo(TABLE_METADATA);
+    assertTableMetadataForCoordinatorTableArePresent();
+  }
+
+  private void assertTableMetadataForCoordinatorTableArePresent() throws Exception {
+    // There is no coordinator table when using JdbcTransaction
+    if (getStorageProperties().getProperty(DatabaseConfig.TRANSACTION_MANAGER, "").equals("jdbc")) {
+      return;
+    }
```

Comment:
This is an implementation specific thing, so I don't think we should put it here.

One possible idea is we can add something like the following method:
```java
  protected boolean hasCoordinatorTables() {
    return true;
  }
```

And then, we can assert the coordinator table metadata when `hasCoordinatorTables()` returns `true`:
```java
  if (hasCoordinatorTables()) {
    assertTableMetadataForCoordinatorTableArePresent();
  }
```

And we can override the method in `JdbcTransactionAdminRepairTableIntegrationTest` and make it return `false`.

What do you think?

Reply:
Thank you. I implemented your suggestion.